### PR TITLE
Wizard recipe: GiNaC-v1.8.6

### PR DESCRIPTION
**Pull request by Fenyutanchan**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: GiNaC
* Version: v1.8.6

### CODE REVIEW

File: G/GiNaC/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+version = v"1.8.6"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://www.ginac.de/ginac-$(version).tar.bz2", "00b320b1116cae5b7b43364dbffb7912471d171f484d82764605d715858d975b")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/ginac-*.*.*/
```
Comment (giordano) [inline G/GiNaC/build_tarballs.jl:15]:
```suggestion
cd $WORKSPACE/srcdir/ginac*/
```

File: G/GiNaC/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    LibraryProduct("libginac", :libginac),
+    ExecutableProduct("viewgar", :viewgar),
+    ExecutableProduct("ginsh", :ginsh)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="Bison_jll", uuid="0f48145f-aea8-549d-8864-7f251ac1e6d0"))
+    Dependency(PackageSpec(name="CLN_jll", uuid="b3974076-79ef-58d3-b5c7-5ef926e97925"))
+    Dependency(PackageSpec(name="flex_jll", uuid="48a596b8-cc7a-5e48-b182-65f75e8595d0"))
```
Comment (giordano) [inline G/GiNaC/build_tarballs.jl:38]: Are you sure flex is a runtime dependency?

File: G/GiNaC/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    Dependency(PackageSpec(name="Python_jll", uuid="93d3a430-8e7c-50da-8e8d-3dfcfb3baf05"))
```
Comment (giordano) [inline G/GiNaC/build_tarballs.jl:39]: What is python used for at all?

```console
% readelf -d libginac.so

Dynamic section at offset 0x2cc8f0 contains 27 entries:
  Tag        Type                         Name/Value
 0x0000000000000001 (NEEDED)             Shared library: [libdl.so.2]
 0x0000000000000001 (NEEDED)             Shared library: [libcln.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libstdc++.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libm.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libgcc_s.so.1]
 0x000000000000000e (SONAME)             Library soname: [libginac.so.11]
```
This doesn't link at all to libpython, so I presume this isn't a runtime dependency?

**Review by giordano — COMMENTED** [review]
You may want to disable building the static library: it's totally useless, and it's huge (over 175 MiB).

**Comment by giordano** [comment]
Also, please ***DO NOT*** force-push without making any actual changes to the source code: we have a caching system here in place which means that the build doesn't happen and the build log is empty, so if I want to go and look at the build log I won't find it in the latest build and I'll need to scavenge it in previous runs, making me waste time.

**Comment by giordano** [comment]
Also, someone should tell the developers that their build system may have some problems:
```console
% readelf -d libginac.so | grep RUNPATH
 0x000000000000001d (RUNPATH)            Library runpath: [RIGIN/../lib:$ORIGIN]
```
https://buildkite.com/julialang/yggdrasil/builds/1858#01873541-0de7-4e0d-b54e-f2e7d212b64a/649-1561
```
[01:22:13] /bin/sh ../libtool  --tag=CXX   --mode=link c++ -std=gnu++11  -g -O2 -version-info 12:5:1  -o libginac.la -rpath /workspace/destdir/lib libginac_la-add.lo libginac_la-archive.lo libginac_la-basic.lo libginac_la-clifford.lo libginac_la-color.lo libginac_la-constant.lo libginac_la-ex.lo libginac_la-excompiler.lo libginac_la-expair.lo libginac_la-expairseq.lo libginac_la-exprseq.lo libginac_la-fail.lo libginac_la-factor.lo libginac_la-fderivative.lo libginac_la-function.lo libginac_la-idx.lo libginac_la-indexed.lo libginac_la-inifcns.lo libginac_la-inifcns_trans.lo libginac_la-inifcns_gamma.lo libginac_la-inifcns_nstdsums.lo libginac_la-inifcns_elliptic.lo libginac_la-integration_kernel.lo libginac_la-integral.lo libginac_la-lst.lo libginac_la-matrix.lo libginac_la-mul.lo libginac_la-ncmul.lo libginac_la-normal.lo libginac_la-numeric.lo libginac_la-operators.lo libginac_la-power.lo libginac_la-registrar.lo libginac_la-relational.lo libginac_la-remember.lo libginac_la-pseries.lo libginac_la-print.lo libginac_la-symbol.lo libginac_la-symmetry.lo libginac_la-tensor.lo libginac_la-utils.lo libginac_la-wildcard.lo parser/libginac_la-parse_binop_rhs.lo parser/libginac_la-parser.lo parser/libginac_la-parse_context.lo parser/libginac_la-default_reader.lo parser/libginac_la-lexer.lo parser/libginac_la-parser_compat.lo polynomial/libginac_la-mod_gcd.lo polynomial/libginac_la-cra_garner.lo polynomial/libginac_la-remainder.lo polynomial/libginac_la-normalize.lo polynomial/libginac_la-upoly_io.lo polynomial/libginac_la-gcd_uvar.lo polynomial/libginac_la-chinrem_gcd.lo polynomial/libginac_la-collect_vargs.lo polynomial/libginac_la-divide_in_z_p.lo polynomial/libginac_la-mgcd.lo polynomial/libginac_la-optimal_vars_finder.lo polynomial/libginac_la-pgcd.lo polynomial/libginac_la-primpart_content.lo -ldl -Wl,-rpath,RIGIN/../lib -Wl,-rpath,/workspace/destdir/lib -Wl,-rpath-link,/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/local/lib -L/workspace/destdir/lib -lcln  -Wl,-rpath -Wl,/workspace/destdir/lib
```
That `RIGIN/../lib` looks terribly wrong. (It took me ***three*** attempts to find the relevant log, as per the message above).

**Comment by Fenyutanchan** [comment]
> Also, please _**DO NOT**_ force-push without making any actual changes to the source code: we have a caching system here in place which means that the build doesn't happen and the build log is empty, so if I want to go and look at the build log I won't find it in the latest build and I'll need to scavenge it in previous runs, making me waste time.

Sorry for that. I clicked the button of `rebase branch` on github.com.

File: G/GiNaC/build_tarballs.jl (hunk 2 above)
Comment (Fenyutanchan) [reply G/GiNaC/build_tarballs.jl:38]: From the logs, it seems that `configure: WARNING: yywrap not found; giving up on flex` forbids the flex.
I'm checking what was wrong.

Comment (Fenyutanchan) [reply G/GiNaC/build_tarballs.jl:38]: Actually, I found the `yywrap` in the `libfl.so`:
```shell
readelf -s ${WORKSPACE}/destdir/lib/libfl.so
```
with
```log
...
    54: 0000000000000550     1 FUNC    GLOBAL DEFAULT    6 _init
    55: 0000000000000740     6 FUNC    GLOBAL DEFAULT    8 yywrap
    56: 0000000000000000     0 NOTYPE  WEAK   DEFAULT  UND __deregister_frame_info
...
```

**Comment by Fenyutanchan** [comment]
> Also, someone should tell the developers that their build system may have some problems:
> 
> ```
> % readelf -d libginac.so | grep RUNPATH
>  0x000000000000001d (RUNPATH)            Library runpath: [RIGIN/../lib:$ORIGIN]
> ```
> 
> https://buildkite.com/julialang/yggdrasil/builds/1858#01873541-0de7-4e0d-b54e-f2e7d212b64a/649-1561
> 
> ```
> [01:22:13] /bin/sh ../libtool  --tag=CXX   --mode=link c++ -std=gnu++11  -g -O2 -version-info 12:5:1  -o libginac.la -rpath /workspace/destdir/lib libginac_la-add.lo libginac_la-archive.lo libginac_la-basic.lo libginac_la-clifford.lo libginac_la-color.lo libginac_la-constant.lo libginac_la-ex.lo libginac_la-excompiler.lo libginac_la-expair.lo libginac_la-expairseq.lo libginac_la-exprseq.lo libginac_la-fail.lo libginac_la-factor.lo libginac_la-fderivative.lo libginac_la-function.lo libginac_la-idx.lo libginac_la-indexed.lo libginac_la-inifcns.lo libginac_la-inifcns_trans.lo libginac_la-inifcns_gamma.lo libginac_la-inifcns_nstdsums.lo libginac_la-inifcns_elliptic.lo libginac_la-integration_kernel.lo libginac_la-integral.lo libginac_la-lst.lo libginac_la-matrix.lo libginac_la-mul.lo libginac_la-ncmul.lo libginac_la-normal.lo libginac_la-numeric.lo libginac_la-operators.lo libginac_la-power.lo libginac_la-registrar.lo libginac_la-relational.lo libginac_la-remember.lo libginac_la-pseries.lo libginac_la-print.lo libginac_la-symbol.lo libginac_la-symmetry.lo libginac_la-tensor.lo libginac_la-utils.lo libginac_la-wildcard.lo parser/libginac_la-parse_binop_rhs.lo parser/libginac_la-parser.lo parser/libginac_la-parse_context.lo parser/libginac_la-default_reader.lo parser/libginac_la-lexer.lo parser/libginac_la-parser_compat.lo polynomial/libginac_la-mod_gcd.lo polynomial/libginac_la-cra_garner.lo polynomial/libginac_la-remainder.lo polynomial/libginac_la-normalize.lo polynomial/libginac_la-upoly_io.lo polynomial/libginac_la-gcd_uvar.lo polynomial/libginac_la-chinrem_gcd.lo polynomial/libginac_la-collect_vargs.lo polynomial/libginac_la-divide_in_z_p.lo polynomial/libginac_la-mgcd.lo polynomial/libginac_la-optimal_vars_finder.lo polynomial/libginac_la-pgcd.lo polynomial/libginac_la-primpart_content.lo -ldl -Wl,-rpath,RIGIN/../lib -Wl,-rpath,/workspace/destdir/lib -Wl,-rpath-link,/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/local/lib -L/workspace/destdir/lib -lcln  -Wl,-rpath -Wl,/workspace/destdir/lib
> ```
> 
> That `RIGIN/../lib` looks terribly wrong. (It took me _**three**_ attempts to find the relevant log, as per the message above).

Okay, I'm missing it. Does 

> Also, someone should tell the developers that their build system may have some problems:
> 
> ```
> % readelf -d libginac.so | grep RUNPATH
>  0x000000000000001d (RUNPATH)            Library runpath: [RIGIN/../lib:$ORIGIN]
> ```
> 
> https://buildkite.com/julialang/yggdrasil/builds/1858#01873541-0de7-4e0d-b54e-f2e7d212b64a/649-1561
> 
> ```
> [01:22:13] /bin/sh ../libtool  --tag=CXX   --mode=link c++ -std=gnu++11  -g -O2 -version-info 12:5:1  -o libginac.la -rpath /workspace/destdir/lib libginac_la-add.lo libginac_la-archive.lo libginac_la-basic.lo libginac_la-clifford.lo libginac_la-color.lo libginac_la-constant.lo libginac_la-ex.lo libginac_la-excompiler.lo libginac_la-expair.lo libginac_la-expairseq.lo libginac_la-exprseq.lo libginac_la-fail.lo libginac_la-factor.lo libginac_la-fderivative.lo libginac_la-function.lo libginac_la-idx.lo libginac_la-indexed.lo libginac_la-inifcns.lo libginac_la-inifcns_trans.lo libginac_la-inifcns_gamma.lo libginac_la-inifcns_nstdsums.lo libginac_la-inifcns_elliptic.lo libginac_la-integration_kernel.lo libginac_la-integral.lo libginac_la-lst.lo libginac_la-matrix.lo libginac_la-mul.lo libginac_la-ncmul.lo libginac_la-normal.lo libginac_la-numeric.lo libginac_la-operators.lo libginac_la-power.lo libginac_la-registrar.lo libginac_la-relational.lo libginac_la-remember.lo libginac_la-pseries.lo libginac_la-print.lo libginac_la-symbol.lo libginac_la-symmetry.lo libginac_la-tensor.lo libginac_la-utils.lo libginac_la-wildcard.lo parser/libginac_la-parse_binop_rhs.lo parser/libginac_la-parser.lo parser/libginac_la-parse_context.lo parser/libginac_la-default_reader.lo parser/libginac_la-lexer.lo parser/libginac_la-parser_compat.lo polynomial/libginac_la-mod_gcd.lo polynomial/libginac_la-cra_garner.lo polynomial/libginac_la-remainder.lo polynomial/libginac_la-normalize.lo polynomial/libginac_la-upoly_io.lo polynomial/libginac_la-gcd_uvar.lo polynomial/libginac_la-chinrem_gcd.lo polynomial/libginac_la-collect_vargs.lo polynomial/libginac_la-divide_in_z_p.lo polynomial/libginac_la-mgcd.lo polynomial/libginac_la-optimal_vars_finder.lo polynomial/libginac_la-pgcd.lo polynomial/libginac_la-primpart_content.lo -ldl -Wl,-rpath,RIGIN/../lib -Wl,-rpath,/workspace/destdir/lib -Wl,-rpath-link,/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/local/lib -L/workspace/destdir/lib -lcln  -Wl,-rpath -Wl,/workspace/destdir/lib
> ```
> 
> That `RIGIN/../lib` looks terribly wrong. (It took me _**three**_ attempts to find the relevant log, as per the message above).

I'm very sorry for that.
I will report this issue to the developers. Thank you very much.
This issue could be solved by using CMake building.

**Pull request merged**